lib/src/client/call.dart

```diff
@@ -241,12 +244,19 @@ class ClientCall<Q, R> implements Response {
       _trailers.complete(metadata);
       // TODO(jakobr): Parse more!
       if (metadata.containsKey('grpc-status')) {
-        final status = int.parse(metadata['grpc-status']);
-        final message = metadata['grpc-message'] == null
-            ? null
-            : Uri.decodeFull(metadata['grpc-message']);
-        if (status != 0) {
-          _responseError(GrpcError.custom(status, message));
+        final status = metadata['grpc-status'];
+        final statusCode = status != null ? int.parse(status) : 0;
```

Comment:
I believe the original author put this check in to ensure we don't try to do `int.parse(null)`. Perhaps this is cleaner?

```dart
final statusCode = int.parse(status ?? '0');
```

Reply:
You mean the check below on line 286 (because there was no check here before your change)?

I think the check there was just another way to write  `_headerMetadata.containsKey('grpc-status')`. It's unclear to me why they choose two inconsistent ways to write it, but I also don't think that this header can even be `null` at this point.

(Well, NNBD migration will reveal the truth).